pkg/api/validation/validation.go

```diff
@@ -2272,6 +2276,12 @@ func ValidatePodSpec(spec *api.PodSpec, fldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, ValidateHostAliases(spec.HostAliases, fldPath.Child("hostAliases"))...)
 	}
 
+	if len(spec.PriorityClassName) > 0 {
```

Comment:
both fields should be prevented from being populated if the priority feature gate is not on... we don't want latent data allowed in, then acted on in a later release when the feature is enabled